Src/PerformanceCollector/Filtering.Shared/DocumentStream.cs

```diff
@@ -190,6 +123,31 @@ public bool CheckFilters(TraceTelemetry document, out CollectionConfigurationErr
 
             return leastOneConjunctionGroupPassed;
         }
+        
+        private static bool CheckFiltersGeneric<TTelemetry>(TTelemetry document, FilterConjunctionGroup<TTelemetry> filterGroup, List<CollectionConfigurationError> errorList)
+        {
+            try
```

Comment:
Can you please refactor this to have a Boolean variable and a single return statement? Multiple returns makes it harder to debug.